articlequality/feature_lists/tests/test_enwiki.py

```diff
@@ -90,3 +90,69 @@ def test_paragraphs_without_refs_total_length():
     """
     assert solve(enwiki.paragraphs_without_refs_total_length,
                  cache={revision_text: text}) == 114
+
+
+def test_all_images():
+    text = """
+        <gallery widths="125px" heights="154px" perrow="4" caption="Studies">
+        File:Alexander Andrejewitsch Iwanow 003.jpg|Study of the head of John
+        File:Ivanov Head of Christ 1824 gtg 17647.jpg|Ivanov head of Christ
+        </gallery>
+
+        [[File:image.jpg|options|caption]]
+
+        {{wide image|Helsinki z00.jpg|400px|
+        |alt=Panorama of city with mixture of five to ten story buildings}}
+
+        {{Panorama
+        |image   = File:Ushuaia_panorama_from_seaside_big.jpg}}
+
+        {{tall image|Polarisation_(Circular).svg}}
+
+        {{scalable image|Dew on grass Luc Viatour.jpg}}
+
+        {{Panorama 2
+        | Image = image file name (don't need Files:)}}
+
+        {{multiple image
+         | align = left
+
+         | image1 = Frecklesmule.jpg
+         | width1 = 143
+         | alt1 = A mule
+         | link1 = Mule
+         | caption1 = A mule<br />(骡子 ''luózi'')}}
+
+        {{Image array
+        | perrow = 2
+        | width = 140
+        | height = 140
+        | border-width = 2
+        | image1 = On The Streets of Vilnius (5984257911).jpg
+        | alt1 = alt1 | text1 = text1 | link1 = Vilnius}}
+
+
+        {{Gallery
+        |title=Cultural depictions of George Washington
+        |width=160 | height=170
+        |align=center
+        |footer=Example 1
+        |File:Federal Hall NYC 27.JPG
+        |alt1=Statue facing a city building with Greek columns
+        and huge U.S. flag
+        |Statue of Washington outside [[Federal Hall]] in [[New York City]]}}
+
+        {{image frame|content={{Photomontage
+        | photo1a = Trujillo Perú PZ.jpg{{!}}Freedom Monument, in Main Square
+        of Trujillo city
+        | photo2a = Húsares.jpg{{!}}"Húsares de Junín" avenue}}
+
+        <imagemap>
+        Image:PolierMartinWombwellZoffany.jpg|thumb|200px|Colonel Antoine
+        Polier
+        rect 269 140 344 305 [[Claude Martin]]
+        rect 124 147 181 298 [[Antoine Polier|Antoine-Louis Polier]]
+        desc none
+        </imagemap>
+        """
+    assert solve(enwiki.all_images, cache={revision_text: text}) == 14
```

Comment:
It would be good to test each sub-feature separately.

Reply:
Noted